src/network/netdev/vxlan.c

```diff
@@ -416,6 +426,10 @@ static int netdev_vxlan_verify(NetDev *netdev, const char *filename) {
                                               "The local address cannot be '%s' when Independent= is enabled, ignoring.",
                                               strna(netdev_local_address_type_to_string(v->local_type)));
 
+        if (v->vnifilter && !v->external) {
```

Comment:
Please merge the condition in the above. Also, please warn if VNI= is specified but External= is also enabled:
```c
if (v->external) {
        if (v->vni <= VXLAN_VID_MAX)
                log_netdev_warning(...);
} else {
        if (v->vni > VXLAN_VID_MAX)
                return log_netdev_warning_errno(...);

        if (v->vnifilter)
                log_netdev_warning(...);
}
```